qemu/qemu.go

```diff
@@ -646,6 +647,400 @@ const (
+
+	// HardDiskTranslationLba lba translation mode
+	HardDiskTranslationLba HardDiskTranslationMode = "lba"
+)
+
+const (
+	// HardDiskCylindersMax max number of cylinders
+	HardDiskCylindersMax uint16 = 16383
+
+	// HardDiskHeadsMax max number of heads
+	HardDiskHeadsMax uint8 = 16
+
+	// HardDiskSectorsMax max number of sectors
+	HardDiskSectorsMax uint8 = 63
+)
+
+// Valid returns true if the hard disk geometry is valid, else false
+func (h *HardDiskGeometry) Valid() bool {
+	if h.Cyls < 1 || h.Cyls > HardDiskCylindersMax {
+		return false
+	}
+
+	if h.Heads < 1 || h.Heads > HardDiskHeadsMax {
+		return false
+	}
+
+	if h.Secs < 1 || h.Secs > HardDiskSectorsMax {
+		return false
+	}
+
+	if h.Trans != "" {
+		switch h.Trans {
+		case HardDiskTranslationAuto, HardDiskTranslationLba, HardDiskTranslationNone:
+			break
+		default:
+			return false
+		}
+	}
+
+	return true
+}
+
+// DriveConnection defines where is connected the drive
+type DriveConnection struct {
+	// Bus where the drive is connected
+	Bus string
+
+	// Unit id where the drive is connected
+	Unit uint
+}
+
+// DriveRWErrors specifies which action to take on write and read errors
+type DriveRWErrors struct {
+	// Werror specifies which action to take on write errors
+	// Available values are: ignore, stop, report, enospc
+	Werror DriveRWErrorsAction
+
+	// Rerror specifies which action to take on read errors
+	// Available values are: ignore, stop, report, enospc
+	Rerror DriveRWErrorsAction
+}
+
+// DriveRWErrorsAction is action to take on rw errors
+type DriveRWErrorsAction string
+
+const (
+	// DriveRWErrorsIgnore ignores the error
+	DriveRWErrorsIgnore DriveRWErrorsAction = "ignore"
+
+	// DriveRWErrorsStop stops Qemu
+	DriveRWErrorsStop DriveRWErrorsAction = "stop"
+
+	// DriveRWErrorsReport reports the error
+	DriveRWErrorsReport DriveRWErrorsAction = "report"
+
+	// DriveRWErrorsEnospc pauses Qemu only if the host disk is full
+	DriveRWErrorsEnospc DriveRWErrorsAction = "enospc"
+)
+
+// Valid returns true if the drive rw error is valid, else false
+func (d *DriveRWErrors) Valid() bool {
+	switch d.Werror {
+	case DriveRWErrorsIgnore, DriveRWErrorsStop, DriveRWErrorsReport, DriveRWErrorsEnospc:
+		break
+	default:
+		return false
+	}
+
+	switch d.Rerror {
+	case DriveRWErrorsIgnore, DriveRWErrorsStop, DriveRWErrorsReport, DriveRWErrorsEnospc:
+		break
+	default:
+		return false
+	}
+
+	return true
+}
+
+// Drive represents a qemu drive.
+type Drive struct {
+	// File defines which disk image to use with this drive
+	File string
+
+	// If defines on which type on interface the drive is connected
+	// Available types are: ide, scsi, sd, mtd, floppy, pflash, virtio
+	If DriveInterface
+
+	// Connection defines where is connected the drive
+	Connection *DriveConnection
+
+	// Index where is connected the drive is connected
+	Index string
+
+	// Media defines the type of the media
+	// Available types are: disk, cdrom
+	Media DriveMedia
```

Comment:
Do we really want to add `cdrom` support? What happens if I don't fill this field?

Reply:
only `File` is mandatory